src/repr/src/row.rs

```diff
@@ -742,6 +831,39 @@ where
                 }
             }
         }
+        Datum::Range(Range { inner }) => {
+            data.push(Tag::Range.into());
+
+            match inner {
+                None => {
+                    data.push(range::RANGE_EMPTY);
+                }
+                Some(RangeInner { lower, upper }) => {
+                    let mut flags = 0;
+
+                    for (val, set) in [
+                        (lower.bound.is_none(), range::RANGE_LB_INF),
+                        (lower.inclusive, range::RANGE_LB_INC),
+                        (upper.bound.is_none(), range::RANGE_UB_INF),
+                        (upper.inclusive, range::RANGE_UB_INC),
+                    ] {
+                        if val {
+                            flags |= set;
+                        }
+                    }
```

Comment:
With bitflags this would become:
```suggestion
                    flags.set(RangeFlags::LOWER_BOUND_INFINITE, lower.bound.is_none());
                    flags.set(RangeFlags::LOWER_BOUND_INCLUSIVE, lower.inclusive);
                    flags.set(RangeFlags::UPPER_BOUND_INFINITE, upper.bound.is_none());
                    flags.set(RangeFlags::UPPER_BOUND_INCLUSIVE, upper.inclusive);
```